wagtail/admin/templates/wagtailadmin/shared/side_panels/includes/status/privacy.html

```diff
@@ -20,14 +20,19 @@
             {{ block.super }}
         {% endwith %}
     </div>
+    {% trans 'Change privacy' as set_privacy_title %}
+    {% trans 'Choose who can view the live version of this page' as set_privacy_message %}
+    {% trans 'Privacy changes apply to all children of this page too' as privacy_message_text %}
+    {% url 'wagtailadmin_pages:set_privacy' page.id as privacy_url %}
+    {% dialog icon_name='view' id='set-privacy' data_url=privacy_url title=set_privacy_title subtitle=set_privacy_message message_status='critical' message_heading=privacy_message_text %}
+    {% enddialog %}
 {% endblock %}
 
 {% block action %}
     {% page_permissions page as page_perms %}
     {% if page.id and page_perms.can_set_view_restrictions %}
-        {% trans 'Change privacy' as set_privacy_text %}
-        {% url 'wagtailadmin_pages:set_privacy' page.id as privacy_url %}
-        {% include 'wagtailadmin/shared/side_panels/includes/side_panel_button.html' with attr='data-action-set-privacy' data_url=privacy_url text=set_privacy_text %}
+        {% trans 'Change privacy' as set_privacy_title %}
+        {% dialog_toggle class_name='w-bg-transparent w-text-14 w-p-0 w-text-teal-200 hover:w-text-teal-600 w-inline-flex w-justify-center w-transition' dialog_id='set-privacy' attr='data-action-set-privacy' text=set_privacy_title %}
```

Comment:
I’m not a fan of this `attr="arbitrary attributes here"` approach, but I suppose it could do the trick for the time being (and it’s what we had before).

Alternatively I think we could try something with `data_url`, to follow what `{% dialog %}` supports?

```suggestion
        {% dialog_toggle class_name='w-bg-transparent w-text-14 w-p-0 w-text-teal-200 hover:w-text-teal-600 w-inline-flex w-justify-center w-transition' dialog_id='set-privacy' data_url=privacy_url text=set_privacy_title %}
```

What do you think?